packages/sdk/src/syncedStream.ts

```diff
@@ -198,7 +198,9 @@ export class SyncedStream extends Stream implements ISyncedStream {
         const minipoolEvents = Array.from(this.view.minipoolEvents.values())
 
         const lastSnapshotMiniblockNum =
-            miniblock.header.snapshot !== undefined || miniblock.header.snapshotHash !== undefined
+            miniblock.header.snapshot !== undefined ||
+            (miniblock.header.snapshotHash !== undefined &&
+                bin_equal(miniblock.header.snapshotHash, snapshot?.hash))
```

Comment:
please change this to check explicitly for the condition where miniblock.header.snapshotHash is defined and does not match snapshot?.hash and log it.

Reply:
actually this is good